src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -8380,6 +8380,48 @@ internal static string HandleDoubleAndSingleQuote(ref string wordToComplete)
             return quote;
         }
 
+        /// <summary>
+        /// Enumerates quoted and unquoted completion text.
+        /// This makes it easier to handle different variations of completions with consideration of quotes.
+        /// </summary>
+        /// <param name="wordToComplete">The word to complete.</param>
+        /// <param name="possibleCompletionValues">The possible completion values to iterate.</param>
+        /// <param name="toolTipMapping">The optional tool tip mapping delegate.</param>
+        /// <param name="resultType">The optional completion result type. Default is Text.</param>
+        /// <returns></returns>
+        internal static IEnumerable<CompletionResult> EnumerateQuotedAndUnquotedCompletionText(
```

Comment:
How about "GetMatchingResults" or something similar?

Reply:
Thanks @MartinGC94. I think `GetMatchingResults` is simpler. Fixed in b0a9dff8914cb2421f39a1366eea2eaae052fd78.